### DISCUSSION TITLE
fzf plugin: why rg chosen in favour of fd?

### DISCUSSION THREAD

**Discussion by pogopaule** [ohmyzsh/ohmyzsh#10755] [Help] [answered]
The [documentation](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/fzf#fzf_default_command) states:

> If not set, the plugin will try to set it to these, in the order in which they're found:
> 
>     [rg](https://github.com/BurntSushi/ripgrep)
>     [fd](https://github.com/sharkdp/fd)
>     [ag](https://github.com/ggreer/the_silver_searcher)

A few years back the author of rg said that fd is the better choice for finding files. [Source](https://www.reddit.com/r/linux4noobs/comments/egb644/fzf_newcomer_fd_or_ripgrep/fc5li3r/?utm_source=reddit&utm_medium=web2x&context=3)

Wouldn't it be better change the order to fd > rg > ag?

**Comment by carlosala** [comment]
I agree with you, I'll submit a patch soon. `fd` is file-specific so should be faster.

**Comment by carlosala** [comment, marked as answer]
Patch submitted #10757!

**Reply by pogopaule** [reply]
That's what I call a response time :smile:  :+1:

**Reply by mcornella** [reply]
PR merged :) Thanks Carlo
